pkg/registry/generic/registry/store.go

```diff
@@ -321,6 +369,10 @@ func (e *Store) Update(ctx api.Context, name string, objInfo rest.UpdatedObjectI
 		if err := rest.BeforeUpdate(e.UpdateStrategy, ctx, obj, existing); err != nil {
 			return nil, nil, err
 		}
+		delete := e.shouldDelete(ctx, key, obj, existing)
```

Comment:
I thought `existing` is the old object? As to whether check finalizers were non-empty, I replied to your other comment.